### PR TITLE
Scom registers access

### PR DESCRIPTION
**Pull request by IgorBagnucki**

### CODE REVIEW

File: devnotes/scom_access.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Access to SCOM registers
+
+According to information given on the OpenPower-firmware mailing list,
```
Comment (macpijan) [inline devnotes/scom_access.md:3]: Please provide a link to this conversation

File: devnotes/scom_access.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+`Not tested`\
+**qemu**:
+```
+Invalid access at addr <address>, size 8, region '(null)', reason: rejected
+XSCOM read failed at @<address> pcba=0x0001e001
+Invalid access at addr <address>, size 8, region '(null)', reason: rejected
+XSCOM read failed at @<address> pcba=0x0001e002
+```
+
+### Assembly instruction **`ldcix`**
```
Comment (macpijan) [inline devnotes/scom_access.md:54]: Please explain in short what kind of instruction is this and why are we trying it there

File: devnotes/scom_access.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   putScom(target, address, data)
+-> platPutScom(target, address, data)
+-> deviceWrite(target, data, size, DEVICE_SCOM_ADDRESS(address, opMode))
+-> Singleton<Associator>::instance().performOp(WRITE, target, buffer, buflen, accessType, args)
+-> findDeviceRoute(opType = WRITE, devType, accessType)(opType = WRITE, target, buffer, buflen, accessType, addr)
+```
+
+before findDeviceRoute(), procedure has to be registerd using Associator.registerRoute()
+
+/src/usr/fsiscom/fsiscom.C:178 fsiScomPerformOp() is probably a function responsible for reading/writing
```
Comment (macpijan) [inline devnotes/scom_access.md:92]: Provide also a github link to this code

File: devnotes/scom_access.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# Access to SCOM registers
```
Comment (macpijan) [inline devnotes/scom_access.md:1]: Are we trying to access register(s) from: https://github.com/3mdeb/openpower-coreboot-docs/blob/main/devnotes/register_for_SCOM_tests.md

If so, explain that and link to this document

File: devnotes/scom_access.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
```
Comment (macpijan) [inline devnotes/scom_access.md:2]: You could briefly describe at the start of the document what is the goal here and why are we trying this.

File: devnotes/scom_access.md (hunk 4 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md:1]: I have been trying to read different register that has other than 0 value on boot. I guess I should add it to the linked document.

File: devnotes/scom_access.md (hunk 5 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md:2]: Added a description

File: devnotes/scom_access.md (hunk 1 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md]: Added a link

File: devnotes/scom_access.md (hunk 2 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md:59]: Added a description

File: devnotes/scom_access.md (hunk 3 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md]: Added a link

File: devnotes/scom_access.md (hunk 4 above)
Comment (IgorBagnucki) [reply devnotes/scom_access.md:1]: I have added this register to the document https://github.com/3mdeb/openpower-coreboot-docs/pull/10

File: devnotes/scom_access.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+XSCOM read failed at @<address> pcba=0x0001e002
+```
+
+### Assembly instruction **`ldcix`**
+Tested code:
+```
+#include <arch/io.h>
+
+uint64_t buffer;
+asm volatile("ldcix %0, %1, %2" : "=r"(buffer) : "b"(0x800623FC00000000ull), "r"(0xF000F));
```
Comment (macpijan) [inline devnotes/scom_access.md:113]: @IgorBagnucki Please explain in this point why it does not work on 1 CPU

Comment (IgorBagnucki) [reply devnotes/scom_access.md:115]: I have added an explanation

**Review by macpijan — APPROVED** [review]

**Pull request merged**